src/Main.elm

```diff
@@ -0,0 +1,197 @@
+module Main exposing (..)
+
+import Html exposing (Html, div, h2, button, text, a)
+import Html.Attributes exposing (href, id)
+import Html.Events exposing (onClick)
+import Movie exposing (Movie, movieCard)
+import MovieList exposing (..)
+import AppCss.Helpers exposing (class)
+import AppCss exposing (..)
+import Time.Date as Date exposing (date)
+import Random
+import Random.List as RandList
+import Markdown
+
+
+main : Program Never Model Msg
+main =
+    Html.program
+        { init = init
+        , view = view
+        , update = update
+        , subscriptions = subscriptions
+        }
+
+
+
+-- Model
+
+
+type alias Model =
+    { unwatched : List Movie
+    , watched : List Movie
+    , selectedMovie : MovieSelection
+    }
+
+
+type MovieSelection
+    = NotSelected
+    | Selected Movie
+    | NothingToSelect
+
+
+init : ( Model, Cmd Msg )
+init =
+    let
+        watchDate movie =
+            Movie.watchDate movie
+                |> Maybe.withDefault (date 1800 1 1)
+    in
+        { unwatched =
+            MovieList.movies
+                |> List.filter (not << Movie.isWatched)
+                |> List.sortBy .year
+        , watched =
+            MovieList.movies
+                |> List.filter Movie.isWatched
+                |> List.sortWith
+                    (\m1 m2 ->
+                        Date.compare (watchDate m1) (watchDate m2)
+                    )
+        , selectedMovie = NotSelected
+        }
+            ! []
+
+
+
+-- Update
+
+
+type Msg
+    = SelectMovie
+    | MovieSelected ( Maybe Movie, List Movie )
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        SelectMovie ->
+            ( model
+            , Random.generate
+                MovieSelected
+                (RandList.choose model.unwatched)
+            )
```

Comment:
Docs for `!` for reference, although they're slim:

http://package.elm-lang.org/packages/elm-lang/core/5.1.1/Platform-Cmd#!

Interesting note - When you see docs for a function in parens like: `(!)` it means that function is used as an infix operator. You can also write your own:

```elm
(@) : Int -> Int -> String
(@) int1 int2 =
    (toString int1) ++ (toString int2)

-- 3 @ 2 == "32"

```